beam/tools/src/main/java/cz/o2/proxima/beam/tools/groovy/BeamStream.java

```diff
@@ -440,6 +441,11 @@ private void runPipeline(Pipeline pipeline) {
     if (!watchTerminating.get()) {
       watch.interrupt();
     }
+    Optional.ofNullable(errorCaught.getAndSet(null))
+        .ifPresent(
+            t -> {
+              throw new RuntimeException("Failed to execute pipeline", t);
```

Comment:
I don't think it's correct to wrap error inside exception. Eg. in case of OOM, JVM can't properly handle it.

Reply:
That's only a rumor that JVM cannot handle OOME. But you are right, that wrapping Error in Exception can be avoided here, thanks!